webapp/channels/src/components/admin_console/schema_admin_settings.jsx

```diff
@@ -571,6 +577,60 @@ export default class SchemaAdminSettings extends React.PureComponent {
         );
     };
 
+    buildRolesSetting = (setting) => {
+        const {roles} = this.props;
+
+        const values = Object.keys(roles).map((r) => {
```

Comment:
You are passing a new list on every render. Not sure if relevant in this case, but you could consider a `useMemo` here.

Reply:
Can't useMemo because it's a class component. I can look at doing something with refs and closures (or a third party lib), but I think it might be a bit overkill for how infrequently this would be rendered. Lmk what you think